core/src/apps/ethereum/sign_tx.py

```
@@ -337,63 +287,184 @@ async def _handle_known_contract_calls(
+                func_sig = None
+                args = None
     else:
-        # If the function was known but something else (data length) was unexpected,
-        # pretend we did not recognize the function so we fall back to blind signing.
-        # See the approve_avantis test case and ERC-8021.
-        func_sig = None
+        args = []
+
+    if func_sig == APPROVE.func_sig:  # custom treatment of "approve"
+        if payment_req_verifier is not None:
+            raise DataError("Payment Requests not supported for the APPROVE call")
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)